app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -147,7 +148,7 @@ open class HomeActivity : AppCompatActivity() {
         // Toggle off the open_link_in_private_tab pref if we are no longer set as the default browser
         // We do this on a separate thread to alleviate performance issues
         val weakReferenceContext = WeakReference(this)
-        lifecycleScope.launch {
+        lifecycleScope.launch(Dispatchers.IO) {
             val context = weakReferenceContext.get() ?: return@launch
             if (!Browsers.all(context).isDefaultBrowser) {
```

Comment:
This code is [roughly duplicated](https://github.com/mozilla-mobile/fenix/blob/562eb89356e291f1037cde6f87cbfe108488f58e/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt#L39-L41). However, the other time it also records a metric – due to the duplication, I believe there's a bug and our metric is undercounting. Would you mind addressing this too or, if it's too much trouble, filing a follow-up?